docs/cli/classifier.py

```diff
@@ -0,0 +1,155 @@
+                lines = lines[idx:]
+                main_start = idx
+                break
+        for i, line in enumerate(lines[1:]):
+            if line and line[0] not in (" ", "\t") and line.strip() != "":
+                lines = lines[:i]
+                break
+
+    for i, line in enumerate(lines, start=1):
+        matches = pattern.findall(line)
+        for key in matches:
+            if key in allowed_keys:
+                if key not in detailed_results:
+                    detailed_results[key] = {"file": path, "lines": []}
+                if file_path.endswith("main.py"):
+                    detailed_results[key]["lines"].append(str(i + main_start))
+                else:
+                    detailed_results[key]["lines"].append(str(i))
+
+    return detailed_results
+
+
+def move_duplicates_together(data, keep):
+    all_keys = list(data.keys())
+    duplicates = set()
+
+    for i, key1 in enumerate(all_keys):
+        for key2 in all_keys[i + 1 :]:
+            duplicates |= data[key1] & data[key2]
+
+    data[keep] |= duplicates
+
+    for key in all_keys:
+        if key != keep:
+            data[key] -= duplicates
+
+    return data
+
+
+def classify(raw_flags):
+
+    category_set = {"general": set(), "linear": set(), "nn": set()}
+    flags = fetch_option_flags(raw_flags)
+    allowed_keys = set(flag["instruction"] for flag in flags)
+    file_set = fetch_all_files()
+    usage_map = defaultdict(list)
+    collected = {}
+
+    for file_path in file_set:
+        detailed_results = find_config_usages_in_file(file_path, allowed_keys)
+        if detailed_results:
+            usage_map[file_path] = set(detailed_results.keys())
+            for k, v in detailed_results.items():
+                if k not in collected:
+                    collected[k] = []
+                collected[k].append(v)
+
+    for path, keys in usage_map.items():
+        category, path = classify_file_category(path)
+        category_set[category] = category_set[category].union(keys)
+
+    category_set = move_duplicates_together(category_set, "general")
+
+    for flag in flags:
+        for k, v in category_set.items():
+            for i in v:
+                if flag["instruction"] == i:
+                    flag["category"] = k
+        if "category" not in flag:
+            flag["category"] = "general"
+
+    result = {}
+    for flag in flags:
+        if flag["category"] not in result:
+            result[flag["category"]] = []
+
+        result[flag["category"]].append(
+            {"name": flag["name"].replace("--", r"\-\-"), "description": flag["description"]}
+        )
+
+    result["details"] = []
+    for k, v in collected.items():
+        result["details"].append({"name": k, "file": v[0]["file"], "location": ", ".join(v[0]["lines"])})
+        if len(v) > 1:
+            for i in v[1:]:
+                result["details"].append({"name": "", "file": i["file"], "location": ", ".join(i["lines"])})
+
+    return result
```

Comment:
How about simplify the data structure (e.g., unused detailed line numbers) after the spec is decided?

```suggestion
def classify(raw_flags):
    category_set = {"general": set(), "linear": set(), "nn": set()}
    flags = fetch_option_flags(raw_flags)
    allowed_keys = set(flag["instruction"] for flag in flags)
    file_set = fetch_all_files()
    
    for file_path in file_set:
        find_config_usages_in_file(file_path, allowed_keys, category_set)
    
    category_set = move_duplicates_together(category_set)

    result = defaultdict(list)
    for flag in raw_flags:
        for category, keys in category_set.items():
            for key in keys:
                if key in flag["name"]:
                    result[category].append(flag)

    return result
```